src/main/java/com/minecolonies/coremod/colony/ColonyView.java

```diff
@@ -205,8 +210,17 @@ public static void serializeNetworkData(@NotNull final Colony colony, @NotNull f
 
         if (colony.getRequestManager() != null && (colony.getRequestManager().isDirty() || hasNewSubscribers))
         {
+            final int preSize = buf.writerIndex();
+            final int preState = buf.readerIndex();
             buf.writeBoolean(true);
             ByteBufUtils.writeTag(buf, colony.getRequestManager().serializeNBT());
+            final int postSize = buf.writerIndex();
+            if ((postSize - preSize) >= MAX_BYTES_NBTCOMPOUND)
+            {
+                colony.getRequestManager().reset();
+                buf.setIndex(preState, preSize);
+                buf.writeBoolean(false);
+            }
```

Comment:
Looks good to me

Reply:
Please run a reset here instead of just not sending it.
You can do this since this is the writing server side.

Reply:
Oh yeah, you can reset and then send, the effect should be immediate anyway.

Reply:
Yep